src/main/scala/com/mozilla/telemetry/pings/Ping.scala

```diff
@@ -0,0 +1,193 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.pings
+
+import java.sql.Timestamp
+
+import com.mozilla.telemetry.pings.Meta._
+import org.joda.time.Months
+import org.joda.time.format.DateTimeFormat
+import org.json4s.JValue
+
+trait Ping {
+  val meta: Meta
+
+  def getExperiments: Array[(Option[String], Option[String])]
+
+  def getVersion: Option[String]
+
+  def getDisplayVersion: Option[String]
+
+  def getOsName: Option[String]
+
+  def getOsVersion: Option[String]
+
+  def getArchitecture: Option[String]
+
+  protected def getRawBuildId: Option[String]
+
+  // TODO: check if we can use appBuildId for all pings
```

Comment:
We should be able to (for telemetry pings), since it's part of the edge server spec [0].

[0] https://docs.telemetry.mozilla.org/concepts/pipeline/http_edge_spec.html#postput-request

Reply:
I'll postpone this for now, there are some crash pings that have these fields different.